src/App.tsx

```diff
@@ -1,93 +1,180 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+
+import { ErrorOnPage } from './components/ErrorOnPage';
+
+import { FilterBy } from './utils/FilterBy';
+import { Todo } from './types/Todo';
+
+import { getFilteredTodos } from './utils/NewfilterTodos';
+
+import { ErrorMessages } from './types/ErrorNessages';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const storedTodos = localStorage.getItem('todos');
+    return storedTodos ? JSON.parse(storedTodos) : [];
+  });
```

Comment:
just as recommendation - you can write a custom useLocalStorage hook
https://designcode.io/react-hooks-handbook-uselocalstorage-hook